process_pr.py

```diff
@@ -1466,86 +1470,194 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
+                        bot_cache[commit_sha]["squashed"] = True
+
             for commit in all_commits:
                 if commit.sha not in bot_cache:
-                    bot_cache[commit.sha] = {"time": int(commit.commit.committer.date.timestamp())}
+                    bot_cache[commit.sha] = {
+                        "time": int(commit.commit.committer.date.timestamp()),
+                        "squashed": False,
+                    }
                     if len(commit.parents) > 1:
                         bot_cache[commit.sha]["files"] = []
                     else:
                         bot_cache[commit.sha]["files"] = sorted(
                             x["filename"] for x in get_commit(repo.full_name, commit.sha)["files"]
                         )
+
                 elif len(commit.parents) > 1:
                     bot_cache[commit.sha]["files"] = []
 
-                if commit.sha not in bot_cache:
-                    bot_cache[commit.sha] = {
-                        "time": int(commit.commit.committer.date.timestamp()),
-                        "files": sorted(
-                            x["filename"] for x in get_commit(repo.full_name, commit.sha)["files"]
-                        ),
-                    }
-
                 cache_entry = bot_cache[commit.sha]
                 events[datetime.fromtimestamp(cache_entry["time"])].append(
-                    {
-                        "type": "commit",
-                        "value": cache_entry["files"],
-                    }
+                    {"type": "commit", "value": {"files": cache_entry["files"], "sha": commit.sha}}
                 )
 
+            # Inject events from cached commits
+            for commit_sha, cache_entry in commit_cache.items():
+                if cache_entry.get("squashed", False):
+                    print("Adding back cached commit {0}".format(commit_sha))
+                    events[datetime.fromtimestamp(cache_entry["time"])].append(
+                        {
+                            "type": "commit",
+                            "value": {"files": cache_entry["files"], "sha": commit_sha},
+                        }
+                    )
+
     # Process commits and signatures
+    signed_commit_sha = ""
+    flattened_eventlist = []
     for _, eventlist in sorted(events.items()):
-        for event in eventlist:
-            print("Event:", event)
-            if event["type"] == "sign":
-                selected_cats = event["value"]["selected_cats"]
-                ctype = event["value"]["ctype"]
-                if any(x in signing_categories for x in selected_cats):
-                    set_comment_emoji_cache(
-                        dryRun, bot_cache, event["value"]["comment"], repository
+        flattened_eventlist.extend(eventlist)
+
+    for event in flattened_eventlist:
+        print("Event:", event)
+        if event["type"] == "sign":
+            comment = event["value"]["comment"]
```

Comment:
@iarspider , what will happen if `signed_commit_sha` is not set ( e.g we have a `+1` before the commit time which can happen when one force push a change after some one has laready signed the previous commit). Do we need to all the logic for `approved/rejected`  in that case? Can we simply skip such comment? e.g. `if not signed_commit_sha: continue` here

Reply:
ah never mind, I forgot that the removed commit will still be in event loop